src/hooks/useSubgraphTokenRecords.ts

```diff
@@ -127,46 +180,89 @@ export const useTokenRecordsQuery = (
  * @param baseFilter
  * @param earliestDate the earliest date to fetch, in YYYY-MM-DD format
  * @param dateOffset the number of days to fetch in each page/request
- * @param shouldHandleHangingQuery if true, will return results even when there is a hanging query
  * @returns Records grouped by date, or null if still fetching
  */
 export const useTokenRecordsQueries = (
   chartName: string,
-  subgraphUrls: SUBGRAPH_URLS,
-  baseFilter: TokenRecord_Filter,
-  earliestDate: string | null,
-  dateOffset?: number,
-  // eslint-disable-next-line @typescript-eslint/no-inferrable-types
-  shouldHandleHangingQuery: boolean = false,
+  subgraphUrls: SUBGRAPH_URLS | null,
+  _baseFilter: TokenRecord_Filter,
+  _earliestDate: string | null,
+  _dateOffset?: number,
 ): Map<string, TokenRecord[]> | null => {
+  // Cache these props, as they will be passed to the query for each blockchain, and we don't want to propagate non-changes
+  const baseFilter = useRef(_baseFilter);
+  useEffect(() => {
+    if (_baseFilter == baseFilter.current) return;
+
+    baseFilter.current = _baseFilter;
+  }, [_baseFilter]);
+  const earliestDate = useRef(_earliestDate);
+  useEffect(() => {
+    if (_earliestDate == earliestDate.current) return;
+
+    earliestDate.current = _earliestDate;
+  }, [_earliestDate]);
+  const dateOffset = useRef(_dateOffset);
+  useEffect(() => {
+    if (_dateOffset == dateOffset.current) return;
+
+    dateOffset.current = _dateOffset;
+  }, [_dateOffset]);
```

Comment:
I'm surprised useMemo doesn't work better here to memoize these values

Reply:
I'll try that!